### PR TITLE
Struct Type to Avro schema re-write. No tests yet

### PR DESCRIPTION
**Pull request by ShannonHolgate**

### CODE REVIEW

File: src/main/scala/com/databricks/spark/avro/SchemaConverters.scala [hunk 1]
```diff
@@ -95,26 +93,75 @@ private object SchemaConverters {
...
+    }.toList
+
+    Schema.createRecord(structName, null, recordNamespace, false, fields)
+  }
+
+  /**
+  * This function is used to construct field schemas of the avro record.
+  * The data type of each field is used to determine which field schema to create.
+  */
+  private[avro] def convertFieldType(dataType: DataType,
```
Comment (progrexor) [inline src/main/scala/com/databricks/spark/avro/SchemaConverters.scala:120]: Why do we need to have **structName** and **recordNamespace** parameters?

File: src/main/scala/com/databricks/spark/avro/LogicalTypeConverters.scala [hunk 2]
```diff
@@ -29,8 +29,18 @@ import org.apache.spark.sql.types._
 
 private object LogicalTypeConverters {
 
-  private[avro] def convertFieldTypeToLogical(): Unit = {
-
+  /**
+  * Converts a supported Spark SQL datatype to the correct avro schema before applying the
+  * logical type schema
+  */
+  private[avro] def convertDataTypeToLogical(dataType: DataType): Schema = {
```
Comment (progrexor) [inline src/main/scala/com/databricks/spark/avro/LogicalTypeConverters.scala:36]: This is minor, **convertDataTypeToLogical** sounds bit incomplete ... Logical what? Maybe worth to rename?
Same with **toSql(...)** method name.

**Review by progrexor — APPROVED** [review]

File: src/main/scala/com/databricks/spark/avro/SchemaConverters.scala (hunk 1 above)
Comment (ShannonHolgate) [reply src/main/scala/com/databricks/spark/avro/SchemaConverters.scala:120]: This method calls back to `convertStructToAvro` to create nested records which have `structName`'s and namespaces

File: src/main/scala/com/databricks/spark/avro/LogicalTypeConverters.scala (hunk 2 above)
Comment (ShannonHolgate) [reply src/main/scala/com/databricks/spark/avro/LogicalTypeConverters.scala:36]: I agree. However, as we are extending an existing library I felt it more accurate to keep to the naming conventions already in use

**Pull request merged**